File: common/src/main/java/com/codeloom/common/event/SubmissionStatusChangedEvent.java
package com.codeloom.common.event;

import com.codeloom.common.SubmissionStatus;
import java.util.UUID;
import lombok.Builder;

@Builder
public record SubmissionStatusChangedEvent(
        UUID submissionId, UUID userId, long problemId, SubmissionStatus newStatus, SubmissionStatusPayload payload) {}

File: backend/src/main/kotlin/com/codeloom/backend/event/SubmissionStatusChangedEvent.kt
package com.codeloom.backend.event

import com.codeloom.backend.model.SubmissionStatus
import java.util.UUID

data class SubmissionStatusChangedEvent(
    val submissionId: UUID,
    val userId: UUID,
    val problemId: Long,
    val newStatus: SubmissionStatus,
    val payload: SubmissionStatusPayload? = null,
)

data class SubmissionStatusPayload(
    val error: String? = null,
    val testCaseResults: List<TestCaseResult>? = null,
)

data class TestCaseResult(
    val id: UUID,
    val problemId: Long,
    val input: String,
    val expectedOutput: String,
    val stdout: String,
    val stderr: String,
    val executionTimeMs: Long,
    val memoryUsageBytes: Long,
)
